actors/user/src/test/java/org/sunbird/user/UserRoleActorTest.java

```diff
@@ -0,0 +1,181 @@
+
+    TestKit probe = new TestKit(system);
+    ActorRef subject = system.actorOf(props);
+    Response response = getCassandraResponse();
+    when(cassandraOperation.getAllRecords(Mockito.anyString(), Mockito.anyString()))
+        .thenReturn(response);
+
+    Request reqObj = new Request();
+    reqObj.setOperation(ActorOperations.GET_ROLES.getValue());
+    subject.tell(reqObj, probe.getRef());
+
+    Response res = probe.expectMsgClass(duration("10 second"), Response.class);
+    Assert.assertTrue(null != res && res.getResponseCode() == ResponseCode.OK);
+  }
+
+  @Ignore
+  public void testAssignRolesSuccessWithValidRole() {
+
+    TestKit probe = new TestKit(system);
+    ActorRef subject = system.actorOf(props);
+
+    when(BaseMWService.getRemoteRouter(Mockito.anyString())).thenReturn(actorSelection);
+
+    when(actorSelection.resolveOneCS(Duration.create(Mockito.anyLong(), "seconds")))
+        .thenReturn(completionStage);
+    when(RequestRouter.getActor(Mockito.anyString())).thenReturn(actorRef);
+
+    when(interServiceCommunication.getResponse(Mockito.anyObject(), Mockito.anyObject()))
+        .thenReturn(response);
+    when(response.get(Mockito.anyString())).thenReturn(map);
+
+    //        when(interServiceCommunication.getResponse(Mockito.anyObject(),
+    // Mockito.anyObject())).thenReturn(organisation);
+
+    Request reqObj = new Request();
+    List roleLst = new ArrayList();
+    roleLst.add("anyRole");
+    reqObj.put(JsonKey.ROLES, roleLst);
+    reqObj.put(JsonKey.EXTERNAL_ID, "EXTERNAL_ID");
+    reqObj.put(JsonKey.USER_ID, "USER_ID");
+    reqObj.put(JsonKey.HASHTAGID, "HASHTAGID");
+    reqObj.put(JsonKey.PROVIDER, "PROVIDER");
+    reqObj.put(JsonKey.ORGANISATION_ID, "ORGANISATION_ID");
+    reqObj.setOperation(ActorOperations.ASSIGN_ROLES.getValue());
+    subject.tell(reqObj, probe.getRef());
+
+    Response res = probe.expectMsgClass(duration("10000 second"), Response.class);
+    Assert.assertTrue(null != res && res.getResponseCode() == ResponseCode.OK);
+  }
+
+  @Ignore
+  public void testAssignRolesFailure() {
+
+    TestKit probe = new TestKit(system);
+    ActorRef subject = system.actorOf(props);
+    Response response = getCassandraResponse();
+    when(cassandraOperation.getAllRecords(Mockito.anyString(), Mockito.anyString()))
+        .thenReturn(response);
+
+    Request reqObj = new Request();
+    reqObj.setOperation(ActorOperations.GET_ROLES.getValue());
+    subject.tell(reqObj, probe.getRef());
+
+    ProjectCommonException res =
+        probe.expectMsgClass(duration("10 second"), ProjectCommonException.class);
+    Assert.assertTrue(null != res);
+  }
+
+  private Response getCassandraResponse() {
+
+    Response response = new Response();
+    List<Map<String, Object>> list = new ArrayList<>();
+    Map<String, Object> orgMap = new HashMap<>();
+    orgMap.put(JsonKey.ID, "org123");
+    list.add(orgMap);
+    response.put(JsonKey.RESPONSE, list);
+    return response;
+  }
+}
```

Comment:
Please check repository, - this file commit is not supposed to be part of this review.